### PR TITLE
[Behat] Current page resolver

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Page/Admin/Crud/PageWithFormInterface.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+namespace Sylius\Behat\Page\Admin\Crud;
+
+use Sylius\Behat\Page\PageInterface;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+interface PageWitHFormInterface extends PageInterface
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/Crud/PageWithFormInterface.php]: Should be ``PageWithFormInterface`` ;)

File: src/Sylius/Behat/Page/Admin/Crud/UpdatePage.php [hunk 2]
```diff
@@ -49,6 +49,19 @@ public function saveChanges()
     /**
      * {@inheritdoc}
      */
+    public function checkValidationMessageFor($element, $message)
+    {
+        $foundedElement = $this->getElement($element)->getParent()->find('css', '.pointing');
+        if(null === $foundedElement) {
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/Crud/UpdatePage.php]: Missing space after ``if``.

Comment (michalmarcinkowski) [inline src/Sylius/Behat/Page/Admin/Crud/UpdatePage.php]: Missing space

File: src/Sylius/Behat/Page/Admin/Crud/UpdatePage.php [hunk 3]
```diff
@@ -49,6 +49,19 @@ public function saveChanges()
...
+            throw new ElementNotFoundException(sprintf('Element %s was not found, so asserting its validation message was not possible.', $element));
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/Crud/UpdatePage.php]: Wdyt about ``Element %s has not been found, so asserting its validation message is not possible.``?

File: src/Sylius/Behat/Service/CurrentPageResolver.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+        if (false !== strpos($routeParameters['_route'], 'create')) {
+            return $createPage;
+        }
+
+        if (false !== strpos($routeParameters['_route'], 'update')) {
+            return $updatePage;
+        }
+
+        throw new \LogicException('Route name does not have any of update or create key word, so matcher was not able to match proper page');
```
Comment (Zales0123) [inline src/Sylius/Behat/Service/CurrentPageResolver.php]: ``'Route name does not have any of "update" or "create" keyword, so matcher was unable to match proper page.'``?

File: src/Sylius/Behat/Service/CurrentPageResolver.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        $routeParameters = $this->urlMatcher->match($this->session->getCurrentUrl());
+
+        if (false !== strpos($routeParameters['_route'], 'create')) {
+            return $createPage;
+        }
+
+        if (false !== strpos($routeParameters['_route'], 'update')) {
+            return $updatePage;
+        }
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Service/CurrentPageResolver.php]: Why not:
```php
public function getCurrentPageWithForm(array $pages)
{
    foreach ($pages => $page) {
        if ($page->isOpen()) {
            return $page;
        }
    }
    throw new \LogicException(...);
}
```

File: src/Sylius/Behat/spec/Service/CurrentPageResolverSpec.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    function it_returns_create_page_if_current_route_name_contains_create_word(
+        Session $session,
+        CreatePageInterface $createPage,
+        UpdatePageInterface $updatePage,
+        UrlMatcherInterface $urlMatcher
+    ) {
+        $session->getCurrentUrl()->willReturn('https://sylius.com/resource/create');
+        $urlMatcher->match('https://sylius.com/resource/create')->willReturn(['_route' => 'sylius_resource_create']);
```
Comment (Zales0123) [inline src/Sylius/Behat/spec/Service/CurrentPageResolverSpec.php]: ``'https://sylius.com/resource/create'`` -> ``'https://sylius.com/resource/new'``? :smile:

File: src/Sylius/Behat/Page/Admin/Crud/PageWithFormInterface.php (hunk 1 above)
Comment (Arminek) [inline src/Sylius/Behat/Page/Admin/Crud/PageWithFormInterface.php]: `FormAwareInterface` ?

**Comment by pjedrzejewski** [comment]
Nice work Łukasz, thanks! :+1: :D

**Pull request merged**